core/src/test/java/org/apache/iceberg/TestPartitionStatsUtil.java

```diff
@@ -370,17 +371,17 @@ public void testPartitionStatsWithSchemaEvolution() throws Exception {
             snapshot2.snapshotId()));
   }
 
-  private static PartitionData partitionData(Types.StructType partitionType, String c2, String c3) {
-    PartitionData partitionData = new PartitionData(partitionType);
-    partitionData.set(0, c2);
-    partitionData.set(1, c3);
-    return partitionData;
+  private static StructLike partitionData(Types.StructType partitionType, String c2, String c3) {
+    GenericRecord record = GenericRecord.create(partitionType);
```

Comment:
Stores the `Record` instead of `PartitionData` in the core logic. So, testcase is updated to expect `Record`.